### PR TITLE
add 2/3d adaptive_avg_pooling

### PR DESCRIPTION
**Pull request by baiyfbupt**
related to https://github.com/PaddlePaddle/Paddle/pull/26369

### CODE REVIEW

**Comment by baiyfbupt** [comment]
image
image

**Comment by baiyfbupt** [comment]
image
image

**Comment by baiyfbupt** [comment]
image
image

**Comment by baiyfbupt** [comment]
image
image

File: doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+参数
+:::::::::
+    - **output_size** (int|list|turple): 算子输出特征图的尺寸，如果其是list或turple类型的数值，必须包含两个元素，H和W。H和W既可以是int类型值也可以是None，None表示与输入特征尺寸相同。
+    - **data_format** (str): 输入和输出的数据格式，可以是"NCHW"和"NHWC"。N是批尺寸，C是通道数，H是特征高度，W是特征宽度。默认值："NCHW"。
+    - **name** (str，可选): 操作的名称(可选，默认值为None）。更多信息请参见 :ref:`api_guide_Name`。
+
+形状
+:::::::::
+    - **x** (Tensor): 当前算子的输入, 是一个4维Tensor。 其数据类型为float16, float32, float64, int32或int64.
+
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst:34]: 形状描述有误

File: doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        #         for j in range(n):
+        #             hstart = floor(i * H / m)
+        #             hend = ceil((i + 1) * H / m)
+        #             wstart = floor(i * W / n)
+        #             wend = ceil((i + 1) * W / n)
+        #             output[:, :, i, j] = avg(input[:, :, hstart: hend, wstart: wend])
+        #
+        import paddle
+        paddle.disable_static()
+        input_data = np.random.rand(2, 3, 32, 32)
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst:67]: 可否用常数 而不是random的变量？

File: doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool3d_cn.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    wend &= ceil((k + 1) * W_{in} / W_{out})
+
+    Output(i ,j, k) &= \frac{sum(Input[dstart:dend, hstart:hend, wstart:wend])}{(dend - dstart) * (hend - hstart) * (wend - wstart)}
+ 
+
+
+参数
+:::::::::
+    - **x** (Tensor): 当前算子的输入, 是一个5维Tensor。 其数据类型为float16, float32, float64, int32或int64.
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool3d_cn.rst]: 5维Tensor -> 5-D Tensor

File: doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool2d_cn.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    wstart &= floor(j * W_{in} / W_{out})
+
+    wend &= ceil((j + 1) * W_{in} / W_{out})
+
+    Output(i ,j) &= \frac{sum(Input[hstart:hend, wstart:wend])}{(hend - hstart) * (wend - wstart)}
+
+
+参数
+:::::::::
+    - **x** (Tensor): 当前算子的输入, 是一个4维Tensor。 其数据类型为float16, float32, float64, int32或int64.
```
Comment (TCChenlong) [inline doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool2d_cn.rst]: 4维Tensor -> 4-D Tensor

**Review by TCChenlong — APPROVED** [review]
LGTM

File: doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst (hunk 1 above)
Comment (baiyfbupt) [reply doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst:34]: fixed

File: doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst (hunk 2 above)
Comment (baiyfbupt) [reply doc/fluid/api_cn/nn_cn/AdaptiveAvgPool2d_cn.rst:67]: 已沟通解决

File: doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool3d_cn.rst (hunk 3 above)
Comment (baiyfbupt) [reply doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool3d_cn.rst]: fixed

File: doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool2d_cn.rst (hunk 4 above)
Comment (baiyfbupt) [reply doc/fluid/api_cn/nn_cn/functional_cn/adaptive_avg_pool2d_cn.rst]: fixed

**Pull request merged**